pyproject.toml

```diff
@@ -57,3 +57,38 @@ nitropy = "pynitrokey.cli:main"
 
 [tool.isort]
 profile = "black"
+
+[tool.mypy]
+mypy_path = "pynitrokey/stubs"
+
+# enable strict checks for new code
+[[tool.mypy.overrides]]
+module = [
+    "pynitrokey.cli.nk3.*",
+    "pynitrokey.nk3.*",
+    "pynitrokey.updates.*",
+]
+disallow_untyped_defs = true
+
+# pynitrokey.nethsm.client is auto-generated
+[[tool.mypy.overrides]]
+module = "pynitrokey.nethsm.client.*"
+ignore_errors = true
+
+# libraries without annotations
+[[tool.mypy.overrides]]
+module = [
+    "cbor.*",
+    "cffi.*",
+    "click.*",
+    "ecdsa.*",
+    "intelhex.*",
+    "nacl.*",
+    "nkdfu.*",
+    "ruamel.*",
+    "serial.*",
+    "urllib3.*",
+    "usb.*",
+    "usb1.*",
```

Comment:
Q: do we use two different USB libraries here (e.g. through dependency)?

Reply:
In pynitrokey itself, we use `usb` (= `pyusb` package) for fido2 and start and `usb1` (= `libusb1` package) for pro and storage.  Also, our `nkdfu` dependency requires `usb1` while `pyocd`/`spsdk` requires `usb`.

<details>

```
$ grep "import usb" -r pynitrokey
pynitrokey/fido2/__init__.py:import usb
pynitrokey/fido2/dfu.py:import usb._objfinalizer
pynitrokey/fido2/dfu.py:import usb.core
pynitrokey/fido2/dfu.py:import usb.util
pynitrokey/cli/pro.py:    import usb1
pynitrokey/cli/storage.py:import usb1
pynitrokey/start/usb_strings.py:import usb
pynitrokey/start/gnuk_token.py:import usb
pynitrokey/start/upgrade_by_passwd.py:    import usb
```

```
$ grep "from usb" -r pynitrokey
pynitrokey/cli/start.py:from usb.core import USBError
```

```
$ pipdeptree --reverse --packages libusb1,pyusb
libusb1==3.0.0
  - nkdfu==0.1 [requires: libusb1>=1.9.3]
    - pynitrokey==0.4.21 [requires: nkdfu]
pyusb==1.2.1
  - pynitrokey==0.4.21 [requires: pyusb]
  - pyocd==0.31.0 [requires: pyusb>=1.2.1,<2.0]
    - spsdk==1.6.3 [requires: pyocd>=0.28.3,<=0.31.0]
      - pynitrokey==0.4.21 [requires: spsdk>=1.6.0,<1.7.0]
```
</details>